core/drand_control.go

```diff
@@ -591,14 +591,61 @@ func (d *Drand) getPhaser(timeout uint32) *dkg.TimePhaser {
 	})
 }
 
+func nodesContainAddr(nodes []*key.Node, addr string) bool {
+	for _, n := range nodes {
+		if n.Address() == addr {
+			return true
+		}
+	}
+	return false
+}
+
+// nodeUnion takes the union of two sets of nodes
+func nodeUnion(a, b []*key.Node) (out []*key.Node) {
+	out = append(out, a...)
+
+OUTER:
+	for _, n := range b {
+		for _, e := range a {
+			if n.Address() == e.Address() {
+				continue OUTER
```

Comment:
what's the purpose of this ??
Why simply not continuing is not enough ?

Reply:
continue will move to the next iteration of the inner loop. we want to move to the next iteration of the outer loop if there's a match

Reply:
```go
func nodeUnion(a, b, []*key.Node) []*key.Node {
  out := append([]*key.Node{},a...)
  for _,n := range b {
    if !nodesContainAddr(a,n) {
        out = append(out,n)
    }
  }
  return out
}
```